modules/nlu/src/views/full/entities/ListEntity.tsx

```diff
@@ -53,20 +70,34 @@ export const ListEntityEditor: React.FC<Props> = props => {
       return
     }
 
-    setOccurrences([...occurrences, { name: newOccurrence, synonyms: [] }])
+    dispatch({
+      type: 'setOccurrences',
+      data: { occurrences: [...state.occurrences, { name: newOccurrence, synonyms: [] }] }
+    })
     setNewOccurrence('')
   }
 
   const editOccurrence = (idx: number, occurrence: NLU.EntityDefOccurrence) => {
-    setOccurrences([...occurrences.slice(0, idx), occurrence, ...occurrences.slice(idx + 1)])
+    const occurrences = [...state.occurrences.slice(0, idx), occurrence, ...state.occurrences.slice(idx + 1)]
```

Comment:
```suggestion
    const occurrences = [...state.occurrences.splice(idx, 1)]
```

Reply:
Sorry that won't work. The return of splice is the removed items and it mutates the original array. 

**From mdn: ** The splice() method changes the contents of an array by removing or replacing existing elements and/or adding new elements in place.